Lib/functools.py

```diff
@@ -980,6 +1004,8 @@ def __set_name__(self, owner, name):
             )
 
     def __get__(self, instance, owner=None):
+        import inspect
```

Comment:
Please move the import just before usage, when the attribute name is not in cache yet.
The import instruction is not for free, it requires a lock at least IIRC.
The move allows returning cached results as fast as possible.